tests/servers/test_navigation_server_3d.h

```diff
@@ -330,6 +357,210 @@ TEST_SUITE("[Navigation]") {
+
+		navigation_server->agent_set_map(agent_2, map);
+		navigation_server->agent_set_avoidance_enabled(agent_2, true);
+		navigation_server->agent_set_position(agent_2, Vector3(2.5, 0, 0.5));
+		navigation_server->agent_set_radius(agent_2, 1);
+		navigation_server->agent_set_velocity(agent_2, Vector3(-1, 0, 0));
+		CallableMock agent_2_avoidance_callback_mock;
+		navigation_server->agent_set_avoidance_callback(agent_2, callable_mp(&agent_2_avoidance_callback_mock, &CallableMock::function1));
+
+		CHECK_EQ(agent_1_avoidance_callback_mock.function1_calls, 0);
+		CHECK_EQ(agent_2_avoidance_callback_mock.function1_calls, 0);
+		navigation_server->process(0.0); // Give server some cycles to commit.
+		CHECK_EQ(agent_1_avoidance_callback_mock.function1_calls, 1);
+		CHECK_EQ(agent_2_avoidance_callback_mock.function1_calls, 1);
+		Vector3 agent_1_safe_velocity = agent_1_avoidance_callback_mock.function1_latest_arg0;
+		Vector3 agent_2_safe_velocity = agent_2_avoidance_callback_mock.function1_latest_arg0;
+		CHECK_MESSAGE(agent_1_safe_velocity.x > 0, "agent 1 should move a bit along desired velocity (+X)");
+		CHECK_MESSAGE(agent_2_safe_velocity.x < 0, "agent 2 should move a bit along desired velocity (-X)");
+		CHECK_MESSAGE(agent_1_safe_velocity.z < 0, "agent 1 should move a bit to the side so that it avoids agent 2");
+		CHECK_MESSAGE(agent_2_safe_velocity.z > 0, "agent 2 should move a bit to the side so that it avoids agent 1");
+
+		navigation_server->free(agent_2);
+		navigation_server->free(agent_1);
+		navigation_server->free(map);
+	}
+
+	// This test case uses only public APIs on purpose - other test cases use simplified baking.
+	TEST_CASE("[NavigationServer3D][SceneTree] Server should be able to bake map correctly") {
+		NavigationServer3D *navigation_server = NavigationServer3D::get_singleton();
+
+		// Prepare scene tree with simple mesh to serve as an input geometry.
+		Node3D *node_3d = memnew(Node3D);
+		SceneTree::get_singleton()->get_root()->add_child(node_3d);
+		Ref<PlaneMesh> plane_mesh = memnew(PlaneMesh);
+		plane_mesh->set_size(Size2(10.0, 10.0));
+		MeshInstance3D *mesh_instance = memnew(MeshInstance3D);
+		mesh_instance->set_mesh(plane_mesh);
+		node_3d->add_child(mesh_instance);
+
+		// Prepare anything necessary to bake navigation mesh.
+		RID map = navigation_server->map_create();
+		RID region = navigation_server->region_create();
+		Ref<NavigationMesh> navigation_mesh = memnew(NavigationMesh);
+		navigation_server->map_set_active(map, true);
+		navigation_server->region_set_map(region, map);
+		navigation_server->region_set_navigation_mesh(region, navigation_mesh);
+		navigation_server->process(0.0); // Give server some cycles to commit.
+
+		CHECK_EQ(navigation_mesh->get_polygon_count(), 0);
+		CHECK_EQ(navigation_mesh->get_vertices().size(), 0);
+
+		navigation_server->region_bake_navigation_mesh(navigation_mesh, node_3d);
```

Comment:
since this part of an API is deprecated I need to create a new test case that uses preferred approach

Reply:
For navigation mesh parsing and baking I would only test the NavigationServer3D::parse_source_geometry_data() and NavigationServer3D::bake_from_source_geometry_data() because those have a future, the NavigationMeshGenerator may have the logic for now but when depr will route to them.